pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -57,6 +57,46 @@ func NewListUsersQuery(ds storage.RelationshipTupleReader, opts ...ListUsersQuer
 	return l
 }
 
+func ValidateListUsersRequest(req *openfgav1.ListUsersRequest, typesys *typesystem.TypeSystem) error {
```

Comment:
Suggestion: move this code that is currently in `server.go`

```
for _, ctxTuple := range req.GetContextualTuples().GetTupleKeys() {
		if err := validation.ValidateTuple(typesys, ctxTuple); err != nil {
			return nil, serverErrors.HandleTupleValidateError(err)
		}
	}
```

to this function?

Reply:
Care should be taken not to do the validation of every iteration of the recursive call though.